pkg/state/types.go

```diff
@@ -672,26 +673,59 @@ func (e EmptyInterfaceUsingUnderlyingType) Less(i, j int) bool {
 	return strings.Compare(objI, objJ) == -1
 }
 
-// Helper function to sort nested arrays in a map
-func sortNestedArrays(m map[string]interface{}) map[string]interface{} {
-	sortedMap := make(map[string]interface{})
+// Helper function to get schema for a field name
+func getSchemaForFieldName(schema gjson.Result, fieldName string) gjson.Result {
+	fieldsQuery := "fields.#(" + fieldName + ")." + fieldName
+	result := schema.Get(fieldsQuery)
+	if !result.Exists() {
+		// try shorthand fields
+		shorthandQuery := "shorthand_fields.#(" + fieldName + ")." + fieldName
+		result = schema.Get(shorthandQuery)
+	}
+	return result
+}
+
+// Helper function to determine if we should skip sorting based on schema
+func skipSort(schema gjson.Result) bool {
+	if !schema.Exists() {
+		return false
+	}
+
+	typeResult := schema.Get("type")
+	if !typeResult.Exists() {
+		return false
+	}
+
+	return typeResult.String() == "array"
+}
+
+// Helper function to sort nested arrays in a map referring to schema
+func sortNestedArraysBasedOnSchema(m map[string]interface{}, schema gjson.Result) map[string]interface{} {
```

Comment:
This implementation of sort is slower than previous, by 10x based on benchmark. However, it still takes few microseconds, which will be negligible compared to rest.